ebml_matroska.xml

```diff
@@ -158,8 +158,10 @@
     <documentation lang="en" purpose="definition">Interpreted by the codec as it wishes (using the BlockAddID).</documentation>
     <extension webm="1"/>
   </element>
-  <element name="BlockDuration" path="0*1(\Segment\Cluster\BlockGroup\BlockDuration)" id="0x9B" type="uinteger" default="DefaultDuration" maxOccurs="1">
-    <documentation lang="en" purpose="definition">The duration of the Block (based on TimestampScale). This Element is mandatory when DefaultDuration is set for the track (but can be omitted as other default values). When not written and with no DefaultDuration, the value is assumed to be the difference between the timestamp of this Block and the timestamp of the next Block in "display" order (not coding order). This Element can be useful at the end of a Track (as there is no other Block available), or when there is a break in a track like for subtitle tracks.</documentation>
+  <element name="BlockDuration" path="0*1(\Segment\Cluster\BlockGroup\BlockDuration)" id="0x9B" type="uinteger" default="see note" minOccurs="see note" maxOccurs="1">
+    <documentation lang="en" purpose="definition">The duration of the Block (based on TimestampScale). The BlockDuration Element can be useful at the end of a Track to define the duration of the last frame (as there is no subsequent Block available), or when there is a break in a track like for subtitle tracks.</documentation>
+    <implementation_note note_type="minOccurs">BlockDuration is a MANDATORY Element if the associated TrackEntry stores a DefaultDuration value.</implementation_note>
```

Comment:
Hmmm, part of the reason that I add `minOccurs="see note"` is because the EBML spec says:
> If the minOccurs attribute is not present then that EBML Element has a minOccurs value of 0.

So either we need to say:
> If the minOccurs attribute is not present and that Element has no implementation_note with a note_attribute set to "minOccurs", then that EBML Element has a minOccurs value of 0.

Or, the EBML specification could have a reserved value ("see note" or something else), to note that the implementation_note defines the minOccurs rather than this attribute.